### PR TITLE
[Beam-10428] Consolidate Go SDK katas as a single module

### PR DESCRIPTION
**Pull request by damondouglas**
The final purpose of this pull request is to consolidate [learning/katas/go](learning/katas/go) into a single module.  Currently, this pull request is not complete and not ready to merge but allows for us to view and agree on changes incrementally.

I would like to request the following reviewers:

(R: @lostluck )
(R: @henryken )

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Dataflow | Flink | Samza | Spark | Twister2
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/) | ---
Java | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/)
Python | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/) | ---
XLang | --- | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/) | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/)
Portable | --- | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by damondouglas** [comment]
@lostluck / @henryken not urgent at all but if you are ok with the progress so far, I will continue making similar changes to the rest of the course sections and lessons.

**Comment by henryken** [comment]
Thanks for trying this out @damondouglas.
It looks good to me and I tested it and it works just fine.

It's unfortunate that we have to use lowercase and underscore for the name of the lessons and sections, but I think it's worth a compromise.

**Comment by damondouglas** [comment]
Thank you, Henry. I'll keep going then. I agree with you about the casing preference. @lostluck are you ok with the module name `apache.org/beam/learning/katas`?

**Comment by damondouglas** [comment]
Wonderful I'll keep pressing on throughout the rest of the course.  As always I will test any time I make changes to make sure everything is working.  As for the module name, I was inspired by how the kubernetes project named their module:  https://github.com/kubernetes/kubernetes/blob/master/go.mod#L7

**Comment by lostluck** [comment]
Oh shoot. Hmm. There might be a problem with the module name in that case: vanity package naming requires setup on the vanity site to find it. Sorry that it only just occurred to me.

**Comment by lostluck** [comment]
In particular, given that users won't really ever import these, we should at least use a module path that we'll eventually have access to change. If everything is already within the sub path (that is users are working with their local module paths), then it shouldn't matter what it is, since they're only importing locally rather than from the "caonical" source (the import path proper).

Had my thoughts long ago about a vanity import path for the SDK.
https://docs.google.com/document/d/15roomfiU1cOSfaDdCzH63VGzySNJ71FZLRZ7qlXJoFQ/edit#heading=h.neqld76lz6xg

The short version is that the path should be beam.apache.org/learning/katas/ since that's the URL we need.

**Comment by damondouglas** [comment]
@lostluck / @henryken I changed the go.mod URL to `beam.apache.org/learning/katas`.  I just wanted to check in before I proceeded incrementally changing the rest of the course.

**Comment by lostluck** [comment]
New path LGTM.

**Comment by henryken** [comment]
LGTM too

**Comment by damondouglas** [comment]
@lostluck / @henryken I made a few rounds of incremental commits.  I wanted to continue to check in before moving forward.

**Comment by damondouglas** [comment]
All needed changes have been made.  @lostluck if this is ok, I will update [the Stepik course](https://stepik.org/course/70387) and let you know when I commit the `*-remote.yaml` files to this PR to be merged into master.

File: learning/katas/go/core_transforms/combine/lesson-info.yaml [hunk 1]
```diff
@@ -18,6 +18,6 @@
 #
 
 content:
-- Simple Function
+- simple_function
 - CombineFn
 - Combine PerKey
```
Comment (lostluck) [inline learning/katas/go/core_transforms/combine/lesson-info.yaml:23]: Shouldn't Combine PerKey change to combine_perkey as well?

Comment (damondouglas) [reply learning/katas/go/core_transforms/combine/lesson-info.yaml:23]: I missed those. Ugh.  👀

**Comment by damondouglas** [comment]
@lostluck if all is well, I can update [the Stepik course](https://stepik.org/course/70387) and let you know when I commit the `*-remote.yaml` files to this PR.

**Comment by damondouglas** [comment]
@lostluck I've updated [the Stepik course](https://stepik.org/course/70387) and committed the `*-remote.yaml` files to this PR.  It is ready to merge.

**Comment by lostluck** [comment]
Thanks!
